src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/GenerateWasmBootJson.cs

```diff
@@ -211,13 +223,48 @@ public void WriteBootJson(Stream output, string entryAssemblyName)
                     resourceList = resourceData.runtime;
                 }
                 else if (string.Equals("JSModule", assetTraitName, StringComparison.OrdinalIgnoreCase) &&
-                            string.Equals(assetTraitValue, "JSLibraryModule", StringComparison.OrdinalIgnoreCase))
+                    string.Equals(assetTraitValue, "JSLibraryModule", StringComparison.OrdinalIgnoreCase))
                 {
                     Log.LogMessage(MessageImportance.Low, "Candidate '{0}' is defined as a library initializer resource.", resource.ItemSpec);
-                    resourceData.libraryInitializers ??= new();
-                    resourceList = resourceData.libraryInitializers;
+
+                    if (TargetingNET80OrLater)
+                    {
+                        resourceData.libraryInitializers ??= new TypedLibraryInitializers();
+                        TypedLibraryInitializers libraryInitializers = (TypedLibraryInitializers)resourceData.libraryInitializers;
```

Comment:
Use `as` so we can catch if the type isn't correct for any reason.

Reply:
nit: If it breaks outside the tests then it would be useful to know what type was actually there.

Reply:
It's set above only if `resourceData.libraryInitializer` is null.

Reply:
Which is guarded by `TargetingNET80OrLater`

Reply:
Outdated